File: CD/projeto/APIWeb/gis/src/main/java/pt/isel/ps/gis/hypermedia/siren/components/subentities/Field.java
package pt.isel.ps.gis.hypermedia.siren.components.subentities;

import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class Field {

    public enum Type {
        hidden("hidden"), text("text"), search("search"), tel("tel"), url("url"), email("email"), password("password"),
        datetime("datetime"), date("date"), month("month"), week("week"), time("time"), datetimeLocal("datetimeLocal"),
        number("number"), range("range"), color("color"), checkbox("checkbox"), radio("radio"), file("file"),
        bool("boolean");

        private final String type;

        Type(String type) {
            this.type = type;
        }

        @Override
        public String toString() {
            return type;
        }
    }

    private String name;
    private Type type;
    private Object value;
    public String title;

    public Field(String name, Type type, Object value, String title) {
        this.name = name;
        this.type = type;
        this.value = value;
        this.title = title;
    }

    public String getName() {
        return name;
    }

    public Type getType() {
        return type;
    }

    public Object getValue() {
        return value;
    }
}

File: Android/gis/app/src/main/java/ps/leic/isel/pt/gis/hypermedia/siren/subentities/Field.kt
package ps.leic.isel.pt.gis.hypermedia.siren.subentities

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@JsonIgnoreProperties(ignoreUnknown = true)
class Field(val name: String, val type: Type?, val value: Any?, var title: String?) {

    enum class Type constructor(private val type: String) {
        Hidden("Hidden"), Text("text"), Search("search"), Tel("tel"), Url("url"), Email("email"), Password("password"),
        Datetime("datetime"), Date("date"), Month("month"), Week("week"), Time("time"), DatetimeLocal("datetimeLocal"),
        Number("number"), Range("range"), Color("color"), Checkbox("checkbox"), Radio("radio"), File("file"),
        Bool("boolean");

        override fun toString(): String {
            return type
        }
    }
}
